sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/implementation/StructuredMessageDecoder.java

```diff
@@ -0,0 +1,182 @@
+package com.azure.storage.common.implementation;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.HashMap;
+import java.util.Map;
+
+public class StructuredMessageDecoder {
+    private static final int DEFAULT_MESSAGE_VERSION = 1;
+    private static final int V1_HEADER_LENGTH = 13;
+    private static final int V1_SEGMENT_HEADER_LENGTH = 10;
+    private static final int CRC64_LENGTH = 8;
+
+    private int messageVersion;
+    private int messageLength;
```

Comment:
We should probably keep this a long to match the type that is supported when using structured message. Technically we should even be treating it as an unsigned long for any comparisons as well.